07-http_gordle/final/internal/gordle/game.go

```diff
@@ -0,0 +1,107 @@
+package gordle
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+// Game holds the information we need to get the Feedback of a play.
+type Game struct {
+	solution []rune
+}
+
+// New returns a Game variable, which can be used to Play!
+func New(corpus []string) (*Game, error) {
+	if len(corpus) == 0 {
+		return nil, ErrEmptyCorpus
+	}
+
+	g := &Game{
+		solution: splitToUppercaseCharacters(pickWord(corpus)), // pick a random word from the corpus
+	}
+
+	return g, nil
+}
+
+const (
+	ErrInvalidGuess = gameError("invalid guess length")
+)
+
+// Play runs the game. If the guess is not valid, we return ErrInvalidGuess.
+func (g *Game) Play(guess string) (Feedback, error) {
+	err := g.validateGuess(guess)
+	if err != nil {
+		return Feedback{}, fmt.Errorf("invalid play: %w", err)
```

Comment:
Ensure that the error which reaches the end users reads something like `This guess is not the correct length. Remember, the answer is 6 letters long.`

Seeing `invalid play: expected 10 characters ...` as the end user feels like a bug in the server.